backend/clubs/views.py

```diff
@@ -2979,6 +2981,22 @@ def get_serializer_class(self):
         return UserProfileSerializer
 
 
+class ClubApplicationViewSet(viewsets.ModelViewSet):
+    """
+    list: Retrieve a list of applications of the club.
+
+    get: Retrieve the details for a given application.
+    """
+
+    permission_classes = [IsSuperuser]
+    serializer_class = ClubApplicationSerializer
+    http_method_names = ["get", "post", "put", "patch", "delete"]
+    lookup_field = "name"
```

Comment:
Having the `lookup_field` be the name won't work well, since it is possible for two applications to have the same name and no unique constraint is being enforced.

I would recommend just removing this field and using the model IDs (built-in) for the individual endpoints. Be sure to return the IDs in the serializer so the client knows what to use.

Reply:
If the url is {club_code}/clubapplication/name, would it make sense to look up by name?